### PR TITLE
Fix column rename in same transaction

### PR DESCRIPTION
**Pull request by dentiny**
Closes https://github.com/duckdb/ducklake/issues/740

Key transaction SQL pasted here
```sql
BEGIN
ALTER TABLE ducklake.expression_column_test ADD COLUMN sort_key_2_new VARCHAR;
ALTER TABLE ducklake.expression_column_test RENAME COLUMN sort_key_2_new TO sort_key_3;
COMMIT
```

**Root cause**
When we rename column after adding the column in one single transaction, DuckLake handles them in this way:
- process add column: add `sort_key_2_new`
- process rename column: drop `sort_key_2_new` and add `sort_key_3`
- As a result, `WriteDropColumns` tries to remove `sort_key_2_new`, `WriteNewColumns` tries to add `sort_key_2_new` and `sort_key_3`
- There're always two obvious issues now: `sort_key_2_new` shouldn't appear in columns to drop and add

But why it doesn't surface during when transaction commits?
Checking how DuckLake drops column at transaction commit, it assumes all columns already committed, and we never check check affected rows
https://github.com/duckdb/ducklake/blob/96f8eb86926add08715b150888a413537b815f31/src/storage/ducklake_metadata_manager.cpp#L2252-L2259

**Followup questions**
Why does it surface at post-transaction query instead of commit?
After transaction commits, we don't check the consistency of data and schema, but did so at query.

This PR tries to solve the issue by recording in-transaction columns, so they're processed in-transaction without touching committed schemas.
This PR shamelessly steals the SQL statements attached in the test file, also add a comprehensive SQL tests for NULL/RENAME/DEFAULT operations in single transaction.

How do we handle multiple column rename operation in one transactions?
Only the last update is processed: https://github.com/duckdb/ducklake/blob/6c69da796c95976eefd6e86856a18769afd0c5ac/src/storage/ducklake_transaction.cpp#L1638-L1644

**Discussion**
To surface the bug earlier, I think one thing we could do is to check affected rows for each sql: https://github.com/duckdb/ducklake/discussions/995

### CODE REVIEW

File: test/sql/alter/multi_alter_same_column_transaction.test [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+    (4, 'c', 'val4');
+
+statement ok
+BEGIN
+
+statement ok
+ALTER TABLE ducklake.expression_column_test ADD COLUMN rename_column_1 VARCHAR;
+
+statement ok
+ALTER TABLE ducklake.expression_column_test RENAME COLUMN rename_column_1 TO rename_column_2;
```
Comment (pdet) [inline test/sql/alter/multi_alter_same_column_transaction.test:75]: What if we Add + Rename +Drop?

Comment (pdet) [reply test/sql/alter/multi_alter_same_column_transaction.test:75]: Also, add two columns and remane both?

**Review by pdet — COMMENTED** [review]
Hi @dentiny, thanks for the PR, just two comments on more testing! :-)

File: test/sql/alter/multi_alter_same_column_transaction.test (hunk 1 above)
Comment (dentiny) [reply test/sql/alter/multi_alter_same_column_transaction.test:75]: That's a good point! Test cases added and fixes made.

**Comment by pdet** [comment]
Thanks!
